requirements-test.txt

```diff
@@ -2,14 +2,16 @@ pytest==7.4.4 ; python_version == '3.8'
 pytest==8.2.1 ; python_version >= '3.9'
 numpy>=1.19.5 ; python_version <= '3.9'
 numpy>=1.21.6 ; python_version == '3.10'
-numpy>=1.23.5 ; python_version >= '3.11'
+numpy>=1.23.5 ; python_version == '3.11'
+numpy==2.0.0rc2 ; python_version >= '3.12'
```

Comment:
We can even keep previous line and do not force this dependency. Might be instead just using separate steps?  But it's fine to keep as is.

One thing is that once 2.0 would be released it would be taken by most Python versions. So we have to put upper bound to control what will be tested with 1.x numpy

Reply:
In general yes - with some restrictions based on pandas and scikit-learn versions as well. I think its worth having broader discussion on how this is handled, but probably best for separate PR

Reply:
For this situation specifically, numpy 2.0.0rc2 is not installed if >=1.23.5 is used (1.26 is instead)

Reply:
Should it be >= 2.0 to allow updates there?

Reply:
Good idea, added it